admin/spam/views.py

```diff
@@ -1,9 +1,37 @@
-from django.http import HttpResponse
+from django.shortcuts import render, render_to_response
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
+
+from .serializers import serialize_comments, retrieve_comment
 
 
 def spam_list(request):
```

Comment:
Not sure if you've investigated the class-based views in django but they might lend some benefit to some of these views that end up being similar. These aren't very long or complicated at this point but as you add more apps to admin I would say CBV would allow you to more easily share code. Probably a judgement call at this point but worth at least investigating.